strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/nuget/TempNupkgFileTest.java

```diff
@@ -0,0 +1,183 @@
+        Path packageFilePath = TestCaseWithNugetPackageGeneration.generatePackageFile(baseDirectoryPath, packageId,
+                                                                                      packageVersion,
+                                                                                      (String[]) null/*
+                                                                                                      * dependencyList
+                                                                                                      */);
+
+        // WHEN
+        try (InputStream nupkgInputStream = new BufferedInputStream(Files.newInputStream(packageFilePath));
+                TempNupkgFile nupkgFile = new TempNupkgFile(nupkgInputStream);)
+        {
+            // TODO: Assert that nupkgFile.getHash() equals generated .nupkg.sha512 file content
+            
+            // THEN
+            assertNotNull(nupkgFile.getHash().toString(),
+                          "Hash file created from stream");
+        }
+    }
+
+    /**
+     * Check reading specifications
+     *
+     * @throws Exception
+     *             error during the test
+     */
+    @Test
+    public void testGetNuspecTmpFile()
+        throws Exception
+    {
+        // GIVEN
+        String expectedPackageId = "NUnit";
+        String expectedPackageVersion = "2.5.9.10348";
+
+        Path packageFilePath = TestCaseWithNugetPackageGeneration.generatePackageFile(baseDirectoryPath,
+                                                                                      expectedPackageId,
+                                                                                      expectedPackageVersion,
+                                                                                      (String[]) null/*
+                                                                                                      * dependencyList
+                                                                                                      */);
+
+        // WHEN
+        try (InputStream nupkgInputStream = new BufferedInputStream(Files.newInputStream(packageFilePath));
+                TempNupkgFile nupkgFile = new TempNupkgFile(nupkgInputStream);)
+        {
+            // WHEN
+            Nuspec nuspecFile = nupkgFile.getNuspec();
+            // THEN
+            assertNotNull(nuspecFile, "Package Specification");
+            assertEquals(expectedPackageId, nuspecFile.getId(), "Package ID");
+            assertEquals(Version.parse(expectedPackageVersion), nuspecFile.getVersion(), "Package Version");
+        }
+    }
+
+    /**
+     * @throws IOException
+     *             error read test data
+     * @throws NugetFormatException
+     *             invalid format exception
+     * @throws JAXBException
+     * @throws NoSuchAlgorithmException
+     */
+    @Test
+    public void testReadNupkg()
+        throws IOException,
+        NugetFormatException,
```

Comment:
Cosmetic: Please, re-indent the exceptions list!

Reply:
I think this is your 3rd comment for exceptions after this one, in which you already say you want this change applied across the PR:

https://github.com/strongbox/strongbox/pull/1106#discussion_r265297268